components/motor/ulnstepper/28byj-48.go

```diff
@@ -0,0 +1,478 @@
+			if err1 != nil {
+				return errors.Errorf("failed to set In1 with error in motor (%s)", m.motorName)
+			}
+
+			time.Sleep(minDelayBetweenTicks)
+			err2 := m.in2.Set(ctx, stepSequence[tick][1], nil)
+			if err2 != nil {
+				return errors.Errorf("failed to set In2 with error in motor (%s)", m.motorName)
+			}
+
+			time.Sleep(minDelayBetweenTicks)
+
+			err3 := m.in3.Set(ctx, stepSequence[tick][2], nil)
+			if err3 != nil {
+				return errors.Errorf("failed to set In3 with error in motor (%s)", m.motorName)
+			}
+
+			time.Sleep(minDelayBetweenTicks)
+
+			err4 := m.in4.Set(ctx, stepSequence[tick][3], nil)
+			if err4 != nil {
+				return errors.Errorf("failed to set In4 with error in motor (%s)", m.motorName)
+			}
+		}
+		time.Sleep(time.Duration(m.setStepperDelay(rpm)))
+		m.stepPosition++
+	} else {
+		for tick := len(stepSequence) - 1; tick >= 0; tick-- {
+			err1 := m.in1.Set(ctx, stepSequence[tick][0], nil)
+			if err1 != nil {
+				return errors.Errorf("failed to set In1 with error in motor (%s)", m.motorName)
+			}
+
+			time.Sleep(minDelayBetweenTicks)
+
+			err2 := m.in2.Set(ctx, stepSequence[tick][1], nil)
+			if err2 != nil {
+				return errors.Errorf("failed to set In2 with error in motor (%s)", m.motorName)
+			}
+
+			time.Sleep(minDelayBetweenTicks)
+
+			err3 := m.in3.Set(ctx, stepSequence[tick][2], nil)
+			if err3 != nil {
+				return errors.Errorf("failed to set In3 with error in motor (%s)", m.motorName)
+			}
+
+			time.Sleep(minDelayBetweenTicks)
+
+			err4 := m.in4.Set(ctx, stepSequence[tick][3], nil)
+			if err4 != nil {
+				return errors.Errorf("failed to set In4 with error in motor (%s)", m.motorName)
+			}
+		}
+		time.Sleep(time.Duration(m.setStepperDelay(rpm)))
+		m.stepPosition--
+	}
+
+	return nil
+}
+
+// GoFor instructs the motor to go in a specific direction for a specific amount of
+// revolutions at a given speed in revolutions per minute. Both the RPM and the revolutions
+// can be assigned negative values to move in a backwards direction. Note: if both are negative
+// the motor will spin in the forward direction.
+func (m *uln2003) GoFor(ctx context.Context, rpm, revolutions float64, extra map[string]interface{}) error {
+	if rpm == 0 {
+		rpm = m.rotationPerMinute
+	}
+	ctx, done := m.opMgr.New(ctx)
+	defer done()
+
+	err := m.goForInternal(ctx, rpm, revolutions)
+	if err != nil {
+		return errors.Wrapf(err, "error in GoFor from motor (%s)", m.motorName)
+	}
+
+	if revolutions == 0 {
+		return nil
+	}
+
+	return m.opMgr.WaitTillNotPowered(ctx, time.Millisecond, m, m.Stop)
+}
+
+func (m *uln2003) goForInternal(ctx context.Context, rpm, revolutions float64) error {
+	if revolutions == 0 {
+		revolutions = 1000000.0
+	}
+	var d int64 = 1
+
+	if math.Signbit(revolutions) != math.Signbit(rpm) {
+		d = -1
+	}
+
+	revolutions = math.Abs(revolutions)
+	rpm = math.Abs(rpm) * float64(d)
+
+	if math.Abs(rpm) < 0.1 {
+		m.logger.Info("RPM is less than 0.1 ", rpm)
```

Comment:
1. can this check be higher up? this can be validated at the start of GoFor and GoTo
2. the log statement needs something to make it clearer e.g. "RPM is less than the 0.1 threshold, stopping."